### PR TITLE
Try to fix linux pdf opening again

### LINKED ISSUE #5942: Opening a PDF blocks JabRef (under Ubuntu/Linux)
JabRef 5.0-beta.419--2020-02-12--3859620
Linux 5.4.19-050419-generic amd64 
Java 13.0.2

- [X] I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

1. Click on the symbol to open a PDF linked to a bib entry (or use F4 or use the "Open menu" command from the context menu). 
2. The PDF viewer (in my case MasterPDFEditor is associated with application/pdf) opens.
3. JabRef can only be used after the PDF viewer is closed again (in the screenshot JabRef is not properly rendered because the whole application is blocked).

jabref-3

I first thought this only happens in my i3/Gnome Flashback environment. But switching to plain Ubuntu 19.10 Gnome desktop, exactly the same behaviour can be observed.

This phenomenon didn't appear for me (with highly similar settings) in JabRef 4.x.

I am seeing this exception on the console after closing JabRef, though this might not result from having called an external program (I suppose):
<details>
  <summary>Log File</summary>
  Exception in Application stop method
ERROR FallbackExceptionHandler Uncaught exception occurred in Thread[main,5,main]
 java.lang.RuntimeException: Exception in Application stop method
	at org.jabref.merged.module/com.sun.javafx.application.LauncherImpl.launchApplication1(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalAccessError: class kong.unirest.JsonObjectMapper (in module org.jabref.merged.module) cannot access class com.google.gson.Gson (in module com.google.gson) because module org.jabref.merged.module does not read module com.google.gson
	at org.jabref.merged.module/kong.unirest.JsonObjectMapper.<init>(Unknown Source)
	at org.jabref.merged.module/kong.unirest.Config.<init>(Unknown Source)
	at org.jabref.merged.module/kong.unirest.Unirest.<clinit>(Unknown Source)
	at org.jabref/org.jabref.Globals.stopBackgroundTasks(Unknown Source)
	at org.jabref/org.jabref.JabRefMain.stop(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$10(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(Unknown Source)
	at org.jabref.merged.module/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(Unknown Source)
	at org.jabref.merged.module/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at org.jabref.merged.module/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(Unknown Source)
	... 1 more

</details>

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fixes #5942  and fixes probably #5882

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not: Issue created at <https://github.com/JabRef/user-documentation/issues>.

### CODE REVIEW

File: src/main/java/org/jabref/gui/desktop/os/Linux.java [hunk 1]
```diff
@@ -53,17 +56,18 @@ public void openFileWithApplication(String filePath, String application) throws
         } else {
             openWith = new String[] {"xdg-open"};
         }
-
         String[] cmdArray = new String[openWith.length + 1];
         System.arraycopy(openWith, 0, cmdArray, 0, openWith.length);
         cmdArray[cmdArray.length - 1] = filePath;
-        Process p = Runtime.getRuntime().exec(cmdArray);
+
+        Process process = Runtime.getRuntime().exec(cmdArray);
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/desktop/os/Linux.java]: I guess the whole process starting has to be done in a different thread...

Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/desktop/os/Linux.java]: That's totally weird as to what I read about is that the process is executed and only if you call waitFor the app blocks. https://www.javaworld.com/article/2071275/when-runtime-exec---won-t.html
And the stream gobbler is there to capture the output

**Comment by Siedlerchr** [comment]
Could you please try again with the newest version from this branch?
I forgot one method.

File: src/main/java/org/jabref/gui/desktop/os/Linux.java (hunk 1 above)
Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/desktop/os/Linux.java]: Strange... can you please try the ProcessBuilder, which anyway seems to be the preferred way to start subprocesses. https://docs.oracle.com/javase/6/docs/api/java/lang/ProcessBuilder.html

File: src/main/java/org/jabref/gui/util/StreamGobbler.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    private Consumer<String> consumer;
+
+    public StreamGobbler(InputStream inputStream, Consumer<String> consumer) {
+        this.inputStream = inputStream;
+        this.consumer = consumer;
+    }
+
+    @Override
+    public void run() {
+        new BufferedReader(new InputStreamReader(inputStream)).lines().forEach(consumer);
```
Comment (koppor) [inline src/main/java/org/jabref/gui/util/StreamGobbler.java]: Does this need to be closed? Do we need try-with-resources here? If not, please add a comment why the buffered reader is not closed.

File: src/main/java/org/jabref/gui/util/StreamGobbler.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    }
+}
```
Comment (koppor) [inline src/main/java/org/jabref/gui/util/StreamGobbler.java]: Please add a new line at the end of the file

**Pull request merged**